### ISSUE TITLE
iOS App - Improvements (Battery Usage and Usability Feature)

### ISSUE THREAD

**Issue by R-Nabil** [netbirdio/netbird#2138]
Hi,

Just wanted to share some observation on the iOS app.
1 - Battery drains faster if the NetBird app is left Connected continuously. FYI, I had the same issue in the past with the Wireguard app, and it seemed to be due to DNS going through the VPN rather than staying "local". Could there be some similar possible improvements on the mobile app to help reduce battery usage ?

2 - In my case, I have multiple exit nodes possible, but I want to be able to opt-in rather than them being automatically selected when I connect to netbird . Is there a way for the app to "remember" the last state and start from that ?

Thanks,

**Comment by roenw** [comment]
+1 on battery drain issue - it’s much worse than the Wireguard iOS client with Wireguard DNS. Screenshot shows WG battery drain between 1800-2200 (left side). Netbird battery drain to the right of that. 
![IMG_9811](https://github.com/user-attachments/assets/2b6fc48b-a09e-48ea-a2ae-8593d510a2c0)

**Comment by gerwim** [comment]
Yes, battery drain on iOS is an issue (cannot confirm if the issue is also present on Android). My iPad lost 29% overnight due to the NetBird app still being open.

**Comment by nazarewk** [comment]
Just to let you know we are working on a "lazy connections" feature which should positively affect both data and battery usage issues.

https://github.com/netbirdio/netbird/pull/3379

**Comment by bobmarley2021** [comment]
@nazarewk is there any update on this? I am absolutely thrilled to see that you’ve just released an android TV app and that Apple TV is imminent along with VPN on-demand being on the roadmap. I’m waiting for these things and battery issues to be resolved before I jump from Tailscale!

**Comment by air5551** [comment]
> Just to let you know we are working on a "lazy connections" feature which should positively affect both data and battery usage issues.
> 
> [#3379](https://github.com/netbirdio/netbird/pull/3379)

Note: I am using the Testflight version of netbird

There are 7 devices connected to my network and netbird with lazy connections on still destroys my battery (I have an Iphone SE Gen 3) also some times devices connect to each other when there is no network activity (Atleast by the apps I have installed). 


Although using the stable release the battery life increases a lot
<img width="750" height="1139" alt="Image" src="https://github.com/user-attachments/assets/60cc6a6e-5098-4a66-8d4c-aa389000c12c" />

**Comment by FxRe** [comment]
Has there been any progress regarding this issue? Is it a general problem or specific to some configurations? I’m new to netbird and after setting it up, this was one of the first things I noticed. If I leave my iPhone connected, battery drains incredibly quickly. I suspected it might have something to do with Rosenpass, but disabling it made little difference, if any. I’m happy to provide additional information or logs, not sure what’s needed.

**Comment by air5551** [comment]
> Has there been any progress regarding this issue? Is it a general problem or specific to some configurations? I’m new to netbird and after setting it up, this was one of the first things I noticed. If I leave my iPhone connected, battery drains incredibly quickly. I suspected it might have something to do with Rosenpass, but disabling it made little difference, if any. I’m happy to provide additional information or logs, not sure what’s needed.

This may help, enabling  Lazy Connections in the Dashboard Settings > Clients > Lazy connections, for me atleast it has saved a lot of battery

**Comment by FxRe** [comment]
> This may help, enabling Lazy Connections in the Dashboard Settings > Clients > Lazy connections, for me atleast it has saved a lot of battery

Thanks for the suggestion. I forgot to mention that I did already have lazy connections enabled, I think that might be the default for new installations.

It's odd, sometimes battery use is totally fine for half a day and then suddenly it starts draining like crazy for no apparent reason. It seems to get triggered particularly when network conditions are changing, e.g. switching from wifi to mobile data and back.
